### PR TITLE
Fill in the 2.8 release note for Authorizer

### PR DESCRIPTION
**Pull request by ctan888**
Fill in the 2.8 release note for Authorizer

### CODE REVIEW

File: docs/upgrade.html [hunk 1]
```diff
@@ -20,6 +20,28 @@
 <script id="upgrade-template" type="text/x-handlebars-template">
 
 <h5><a id="upgrade_280_notable" href="#upgrade_280_notable">Notable changes in 2.8.0</a></h5>
+<ul>
+    <li>
+        The 2.8.0 release added a new Authorizer interface introduced in
```
Comment (rajinisivaram) [inline docs/upgrade.html]: added a new method to the Authorizer Interface

File: docs/upgrade.html [hunk 2]
```diff
@@ -20,6 +20,28 @@
...
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-679%3A+Producer+will+enable+the+strongest+delivery+guarantee+by+default">KIP-679</a>.
+        Custom authorizer implementations should consider overriding this default implementation because:
```
Comment (rajinisivaram) [inline docs/upgrade.html]: It will be good to specify the change to authorization (i.e. why we added the new method).

File: docs/upgrade.html [hunk 3]
```diff
@@ -20,6 +20,28 @@
...
+        <ul>
+            <li>
+                1. The default implementation iterates all AclBindings multiple times, without any caching
+                   by principal, host, operation, permission types, and resource types. More efficient
+                   implementations may be added in custom authorizers that directly access cached entries.
```
Comment (rajinisivaram) [inline docs/upgrade.html]: Should be sufficient to provide a summary: e.g. `Custom authorizers should consider providing a more efficient implementation that supports audit logging and any custom configs or access rules`.

**Review by rajinisivaram — COMMENTED** [review]
@d8tltanc Thanks for the PR, left a few comments.

File: docs/upgrade.html (hunk 3 above)
Comment (ijuma) [reply docs/upgrade.html]: Yes, refer to the Authorizer javadocs for detailed reasoning.

**Comment by ctan888** [comment]
Hi @rajinisivaram . Thanks for the comments. I've addressed all of them. Could you take a look and approve it? Thanks.

**Review by rajinisivaram — APPROVED** [review]
@d8tltanc Thanks for the update, merging to trunk and 2.8.

**Pull request merged**